### PR TITLE
feat: load github token from file

### LINKED ISSUE #535: Support for reading GitHub token from a file
Creating a GitHub release currently requires the `GITHUB_TOKEN` environment variable to be set. This is great for CI environments, but less so on a development machine; it means that I either need to set `GITHUB_TOKEN` in my shell profile, in which case it's trivially exposed to every process I run, or use some external system like [direnv] to load it on-demand. By contrast, GitHub's hub tool [stores the OAuth token in `~/.config/hub`][hub], which is a YAML file with mode `0600`. It would be nice if GoReleaser supported something similar.

[direnv]: https://github.com/direnv/direnv
[hub]: https://hub.github.com/hub.1.html#GitHub-OAuth-authentication

### PR DESCRIPTION
**Pull request by caarlos0**
closes #535

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] I have read the **CONTRIBUTING** document.
* [ ] `make ci` passes on my machine.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: docs/020-environment.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ This token should be added to the environment variables as `GITHUB_TOKEN`.
 Here is how to do it with Travis CI:
 [Defining Variables in Repository Settings](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings).
 
+If you don't want to have a globaly available github token on your env, you
```
Comment (elyscape) [inline docs/020-environment.md]: This is probably fine, but I'd probably word it as something like:
>Alternatively, you can provide the GitHub token in a file. GoReleaser will check `~/.config/goreleaser/github_token` by default.

I would also recommend documenting the `env_files` config setting somewhere.

File: pipeline/env/env.go [hunk 2]
```diff
@@ -20,17 +22,49 @@ func (Pipe) String() string {
...
+func loadEnv(env, path string) (string, error) {
+	val := os.Getenv(env)
+	if val != "" {
+		return val, nil
+	}
+	path, err := homedir.Expand(path)
+	if err != nil {
+		return "", err
+	}
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (caarlos0) [reply pipeline/env/env.go]: that's already covered by the ioutil.ReadFile.

in fact, I can remove the os.Stat call :D

**Comment by elyscape** [comment]
>yeah, I don't think it makes a lot of sense for goreleaser, since the user can change the secrets file in the .goreleaser.yml (thus having another token for each project for example)

That makes sense, though as noted in that case you'll need to have `env_files` documented.

>I didn't wanted a yaml file because it would one more thing to parse and having a simple text file looks good enough for me :)

Totally fair! In that case, this looks fine to me.

**Comment by caarlos0** [comment]
nice @elyscape !

Will release this probably today :D

**Pull request merged**